### PR TITLE
Pre view_classes cleanups

### PR DESCRIPTION
**Pull request by phlax**

### CODE REVIEW

File: pootle/apps/pootle_language/models.py [hunk 1]
```diff
@@ -169,18 +169,17 @@ def get_cachekey(self):
...
-                                   .order_by('project__fullname')
-        user_tps = filter(lambda x: x.is_accessible_by(user),
-                          translation_projects)
-
-        return user_tps
+    def get_children_for_user(self, user, select_related=None):
+        tps = self.translationproject_set.for_user(
+            user, select_related=select_related)
+        return filter(
+            lambda x: x.is_accessible_by(user),
```
Comment (ta2-1) [inline pootle/apps/pootle_language/models.py]: `x.project.code` is retrieved in `x.is_accessible_by(user)` method. select_related(`project__code`) should be better.

File: pootle/apps/pootle_project/models.py [hunk 2]
```diff
@@ -402,12 +402,14 @@ def get_cachekey(self):
     # # # /TreeItem
 
     def get_stats_for_user(self, user):
-        self.set_children(self.get_children_for_user(user))
+        self.set_children(
+            self.get_children_for_user(
+                user, select_related=["language__name", "directory"]))
```
Comment (ta2-1) [inline pootle/apps/pootle_project/models.py]: @phlax, you wrote: "i dont think you can do select_related("fk__otherfield")"...

Comment (ta2-1) [inline pootle/apps/pootle_project/models.py]: even more: Project doesn't have language fk.

Comment (phlax) [inline pootle/apps/pootle_project/models.py]: fk == ForeignKey

File: pootle/apps/pootle_project/models.py [hunk 3]
```diff
@@ -402,12 +402,16 @@ def get_cachekey(self):
     # # # /TreeItem
 
     def get_stats_for_user(self, user):
-        self.set_children(self.get_children_for_user(user))
+        self.set_children(
+            self.get_children_for_user(
+                user, select_related=["language"]))
```
Comment (ta2-1) [inline pootle/apps/pootle_project/models.py]: this select related should come from views. language is not necessary for stats.

Comment (phlax) [inline pootle/apps/pootle_project/models.py]: k

File: pootle/apps/pootle_project/models.py [hunk 4]
```diff
@@ -402,12 +402,16 @@ def get_cachekey(self):
...
         return self.get_stats()
 
-    def get_children_for_user(self, user):
+    def get_children_for_user(self, user, select_related=None):
         """Returns children translation projects for a specific `user`."""
-        return self.translationproject_set.for_user(user)
+        return (
+            self.translationproject_set.for_user(user, select_related)
+                                       .select_related("project"))
```
Comment (ta2-1) [inline pootle/apps/pootle_project/models.py]: language can be added here

File: pootle/apps/pootle_project/models.py [hunk 5]
```diff
@@ -403,11 +403,14 @@ def get_cachekey(self):
...
+
         return self.get_stats()
 
-    def get_children_for_user(self, user):
+    def get_children_for_user(self, user, select_related=None):
         """Returns children translation projects for a specific `user`."""
-        return self.translationproject_set.for_user(user)
+        return (
+            self.translationproject_set.for_user(user, select_related)
+                                       .select_related("language"))
```
Comment (ta2-1) [inline pootle/apps/pootle_project/models.py]: @phlax, did you remove project from here?

Comment (phlax) [inline pootle/apps/pootle_project/models.py]: yep - but i might need to readd it - i was just about to check

**Comment by ta2-1** [comment]
@phlax, lgtm if you reckon that project is unnecessary (either add it) and tests pass.

**Pull request merged**